jr-objects/src/main/java/com/fasterxml/jackson/jr/ob/impl/BeanPropertyIntrospector.java

```diff
@@ -142,6 +143,13 @@ private static void _introspect(Class<?> currType, Map<String, PropBuilder> prop
                         name = decap(name.substring(2));
                         _propFrom(props, name).withIsGetter(m);
                     }
+                } else {
+                    // This will allow getters with field name as their getters, like the ones generated by Groovy
+                    boolean isDirectNameGetter = Arrays.stream(currType.getDeclaredFields()).map(Field::getName).anyMatch(f -> f.equals(m.getName()));
```

Comment:
i believe if we get a list of fields & check if there is a field with the same name as current method, then we need to add it as a prop

, this can be optimized, but as of now, its working.

Reply:
I think this capability needs to be behind `Json.Feature` flag since while it makes sense wrt Records (and probably other value types), it can also be unexpected for other users. That is, it changes compatibility.

Other than that it'd make sense to build a map from name to `Field` in loop above, to avoid linear scanning for every potential "non-get-getter".